cmd/juju/resource/deploy.go

```diff
@@ -273,9 +274,17 @@ func unMarshalDockerDetails(data io.Reader) (resources.DockerImageDetails, error
 		return details, errors.Trace(err)
 	}
 
-	if err := json.Unmarshal(contents, &details); err != nil {
-		if err := yaml.Unmarshal(contents, &details); err != nil {
-			return details, errors.Annotate(err, "file neither valid json or yaml")
+	if errJ := json.Unmarshal(contents, &details); errJ != nil {
+		if errY := yaml.Unmarshal(contents, &details); errY != nil {
+			contentType := http.DetectContentType(contents)
+			if strings.Contains(contentType, "text/plain") {
+				// Check first character - `{` means probably JSON
+				if strings.TrimSpace(string(contents))[0] == '{' {
+					return details, errors.Annotate(errJ, "json parsing")
+				}
+				return details, errY
+			}
+			return details, errors.New("expected json or yaml file")
 		}
 	}
```

Comment:
This can be simplified.... check binary/text first and exit early
```
if !text {
 error("expected json or yaml file")
}
... json/yaml unmarshall ...
```
Also maybe make the non text error message more verbose, eg
"expected json or yaml file containing the oci-image registry details, not the actual image"

Reply:
Would rather be safe and attempt to parse it anyway, in case it gets misclassified